src/code/z_room.c

```diff
@@ -605,29 +611,29 @@ s32 func_8009728C(PlayState* play, RoomContext* roomCtx, s32 roomNum) {
         roomCtx->curRoom.segment = NULL;
         roomCtx->status = 1;
 
-        ASSERT(roomNum < play->numRooms, "read_room_ID < game_play->room_rom_address.num", "../z_room.c", 1009);
+        ASSERT(roomNum < play->roomList.num, "read_room_ID < game_play->room_rom_address.num", "../z_room.c", 1009);
 
-        size = play->roomList[roomNum].vromEnd - play->roomList[roomNum].vromStart;
-        roomCtx->unk_34 =
-            (void*)ALIGN16((uintptr_t)roomCtx->bufPtrs[roomCtx->unk_30] - ((size + 8) * roomCtx->unk_30 + 7));
+        size = play->roomList.romFiles[roomNum].vromEnd - play->roomList.romFiles[roomNum].vromStart;
+        roomCtx->roomRequestAddr = (void*)ALIGN16((uintptr_t)roomCtx->bufPtrs[roomCtx->activeBufPage] -
+                                                  ((size + 8) * roomCtx->activeBufPage + 7));
 
         osCreateMesgQueue(&roomCtx->loadQueue, &roomCtx->loadMsg, 1);
-        DMA_REQUEST_ASYNC(&roomCtx->dmaRequest, roomCtx->unk_34, play->roomList[roomNum].vromStart, size, 0,
-                          &roomCtx->loadQueue, NULL, "../z_room.c", 1036);
-        roomCtx->unk_30 ^= 1;
+        DMA_REQUEST_ASYNC(&roomCtx->dmaRequest, roomCtx->roomRequestAddr, play->roomList.romFiles[roomNum].vromStart,
+                          size, 0, &roomCtx->loadQueue, NULL, "../z_room.c", 1036);
+        roomCtx->activeBufPage ^= 1;
 
         return true;
     }
 
     return false;
 }
 
-s32 func_800973FC(PlayState* play, RoomContext* roomCtx) {
+s32 Room_HandleLoadCallbacks(PlayState* play, RoomContext* roomCtx) {
     if (roomCtx->status == 1) {
         if (osRecvMesg(&roomCtx->loadQueue, NULL, OS_MESG_NOBLOCK) == 0) {
             roomCtx->status = 0;
```

Comment:
I've felt very strongly for a while about not liking his name in MM. There are no room callbacks here. At best you can say the scene system uses callbacks with its own processing? But yeah I feel this can be improved.

I also am not a fan of `Room_LeavePrevRoom`, the function handles both leaving the room and tying some loose ends with starting the new room.

Following your example with `Room_StartRoomTransition`, I suggest the following:
`Room_StartRoomTransition` - What you have currently, it starts the swapping process and starts the DMA for the new room
`Room_ProcessRoomTransition` - Waits for the dma to finish, then does the bulk of processing the new room, including running all of the scene commands and a few other tasks
`Room_FinishRoomTransition` - Nulls the previous room pointer, kills actors in the prev room, then does some final initialization for the new room. (Side note: why does this call `Actor_SpawnTransitionActors` again?)

Reply:
just a detail but I think names like ``Room_StartRoomTransition`` are a bit redundant because there's already the ``Room_`` prefix so it's obvious it's for room transitions, ``Room_StartTransition`` sounds better imo (same for the other names)

Reply:
in this case I think the redundancy is not a problem, its just being more specific about what exactly is transitioning. Room is the system name, and there could theoretically be other kinds of transitions in the room system.
But more opinions needed

Reply:
yeah I guess it's just unfortunate the system and the transition type share the same name

Reply:
After even more thought I think I would prefer "Change" over "Transition"
`Room_StartRoomChange`
`Room_ProcessRoomChange`
`Room_FinishRoomChange`

Reply:
Perhaps I probably shouldn't have just blindly copied MM's function names, because the more I look at all of them the less I like them.

I'm starting to dislike `StartRoomTransition`, `ProcessRoomTransition`, `FinishRoomTransition` because there is nothing there to manage a transition between two rooms. Specifically, the first room that is loaded in a new PlayState is created by calling just `StartRoomTransition` followed by `ProcessRoomTransition`. `FinishRoomTransition` is never called since it's actually not needed to "finalize" the room in this case; this is why I feel `LeavePrevRoom` is appropriate. 

To me, this set of functions should be named more like

* `Room_LoadNewRoom`
* `Room_InitializeNewRoom`
* `Room_ExitPrevRoom`/`Room_LeavePrevRoom`


The other function name I want to change is `Room_AllocateAndLoad` to something like `Room_InitializeFirstRoom` or `Room_SetupFirstRoom`, but idk. `Room_InitializeFirstRoom` is weird because the room won't actually be initialized until what is currently `Room_HandleLoadCallbacks` is called.

Reply:
I can get on board with this direction. `Room_LoadNewRoom` makes it sound like its responsible for waiting for the dma, meanwhile that happens in `Room_InitializeNewRoom`. But I dont have any other suggestions then going back to using "start"

Reply:
Actually tbh I still think I like Start/Process/Finish Room Change. `Room_ExitPrevRoom` Still doesnt address my previous concern that the function is doing things for both rooms. And start makes more sense for the reason above. and "process" is broad enough to encompass the dma load time and initialization/processing of the scene commands and other extra bits

Reply:
Well, your point was that `Room_FinishRoomChange` did further initialization of the current room, but I don't find this to be accurate.

If we look at the function again, we can break it down into four key operations

* Invalidates/deletes the previous room
* Kill all non-active actors
   * I believe `Actor_SpawnTransitionActors` is called here as an emergency failsafe. Looking at `func_80031B14` and `Actor_SpawnTransitionActors`, I believe the function shouldn't actually spawn anything in normal circumstances. However in low memory situations, this may allow transition actors that failed to spawn earlier the chance to spawn in memory that was just freed from killed actors, but I haven't tested this in game.
* Update the map system to recognize we're in a new room
* Update the environment echo setting

None of these are directly related to initializing the new room, but rather are for either cleaning up leftovers of the previous room or updating external systems to make them know that the player has fully moved into the next room.

Furthermore, it should be noted that...

* The state of `Room curRoom` never changes during the function.
* The `curRoom` created in `Room_LoadNewRoom` and `Room_InitializeNewRoom` can become the `prevRoom` by the time `Room_ExitPrevRoom` is called, so again it wouldn't be initializing the newly created room.

Reply:
per discussion on Discord i am tossing my vote in for
`Room_RequestRoomChange`
`Room_ProcessRoomChange`
`Room_FinishRoomChange`